core/src/main/kotlin/entity/component/SelectMenuComponent.kt

```diff
@@ -30,6 +28,8 @@ public class SelectMenuComponent(override val data: ComponentData) : Component {
 
     /**
      * The possible options to choose from.
+     *
+     * **Available only in [ComponentType.StringSelect]**
      */
     public val options: List<SelectOption> get() = data.options.orEmpty().map { SelectOption(it) }
```

Comment:
I think this should be deprecated instead and be added to the new `StringSelectComponent` subtype.